File: recursion.py
from time import perf_counter_ns
import sys

# 재귀 깊이 제한 설정
sys.setrecursionlimit(100000)


def getSum(n):
    if n == 1:
        return 1
    return n + getSum(n - 1)


sum = 0

for i in range(10):
    start = perf_counter_ns()

    result = getSum(10000)

    end = perf_counter_ns()

    elapsed_ns = end - start
    elapsed_ms = elapsed_ns / 1000000
    sum += elapsed_ms

    print(f"Iteration #{i + 1} {elapsed_ns}ns ({elapsed_ms}ms)")

print(f"Average: {sum / 10}ms")

File: recursion.c
#include <stdio.h>
#include <time.h>

#define ll long long

int getSum(int n)
{
    if (n == 1)
        return 1;
    return n + getSum(n - 1);
}

int main()
{
    double sum = 0;

    struct timespec start, end;

    for (int i = 0; i < 10; i++)
    {
        timespec_get(&start, TIME_UTC);
        int result = getSum(10000);
        timespec_get(&end, TIME_UTC);

        ll elapsed_ns = (ll)(end.tv_sec - start.tv_sec) * 1000000000ll +
                        (ll)(end.tv_nsec - start.tv_nsec);
        double elapsed_ms = elapsed_ns / 1000000.0;
        sum += elapsed_ms;

        printf("Iteration #%d: %lldns (%lfms)\n", i + 1, elapsed_ns, elapsed_ms);
    }

    printf("Average: %lfms", sum / 10);

    return 0;
}